<a href="https://colab.research.google.com/github/ElijahFreeney/Machine-Learning/blob/main/SKLearn.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
import sklearn as df
import pandas as pd

In [ ]:
df = pd.read_csv('Auto.csv')
print(df.head)

<bound method NDFrame.head of       mpg  cylinders  displacement  horsepower  weight  acceleration  year  \
0    18.0          8         307.0         130    3504          12.0  70.0   
1    15.0          8         350.0         165    3693          11.5  70.0   
2    18.0          8         318.0         150    3436          11.0  70.0   
3    16.0          8         304.0         150    3433          12.0  70.0   
4    17.0          8         302.0         140    3449           NaN  70.0   
..    ...        ...           ...         ...     ...           ...   ...   
387  27.0          4         140.0          86    2790          15.6  82.0   
388  44.0          4          97.0          52    2130          24.6  82.0   
389  32.0          4         135.0          84    2295          11.6  82.0   
390  28.0          4         120.0          79    2625          18.6  82.0   
391  31.0          4         119.0          82    2720          19.4  82.0   

     origin                      

Data exploration
Here, we can see a summary of the three columns: mpg, weight, and year

In [ ]:
df[['mpg', 'weight', 'year']].describe()


mpg       weight        year
count  392.000000   392.000000  390.000000
mean    23.445918  2977.584184   76.010256
std      7.805007   849.402560    3.668093
min      9.000000  1613.000000   70.000000
25%     17.000000  2225.250000   73.000000
50%     22.750000  2803.500000   76.000000
75%     29.000000  3614.750000   79.000000
max     46.600000  5140.000000   82.000000


*   MPG: has a range between 9 to 46.6 and a mean of 23.45.
*   Weight: has a range between 1613 to 5140 and a mean of 2977.
*   Year: has a range between 70 to 79 and a mean of 2977


Let's dive into the data. First checking the data types

In [ ]:
print(df.dtypes)

mpg             float64
cylinders         int64
displacement    float64
horsepower        int64
weight            int64
acceleration    float64
year            float64
origin            int64
name             object
dtype: object


Change column types with and without cat.codes.

In [ ]:
df['cylinders'] = df['cylinders'].astype('category').cat.codes
df['origin'] = df['origin'].astype('category')
print(df.dtypes)



mpg              float64
cylinders           int8
displacement     float64
horsepower         int64
weight             int64
acceleration     float64
year             float64
origin          category
name              object
dtype: object


Delete row's with NA's in any column field

In [ ]:
print(df)
df.dropna()


      mpg  cylinders  displacement  horsepower  weight  acceleration  year  \
0    18.0          4         307.0         130    3504          12.0  70.0   
1    15.0          4         350.0         165    3693          11.5  70.0   
2    18.0          4         318.0         150    3436          11.0  70.0   
3    16.0          4         304.0         150    3433          12.0  70.0   
4    17.0          4         302.0         140    3449           NaN  70.0   
..    ...        ...           ...         ...     ...           ...   ...   
387  27.0          1         140.0          86    2790          15.6  82.0   
388  44.0          1          97.0          52    2130          24.6  82.0   
389  32.0          1         135.0          84    2295          11.6  82.0   
390  28.0          1         120.0          79    2625          18.6  82.0   
391  31.0          1         119.0          82    2720          19.4  82.0   

    origin                       name  
0        1  chevrolet c

mpg  cylinders  displacement  horsepower  weight  acceleration  year  \
0    18.0          4         307.0         130    3504          12.0  70.0   
1    15.0          4         350.0         165    3693          11.5  70.0   
2    18.0          4         318.0         150    3436          11.0  70.0   
3    16.0          4         304.0         150    3433          12.0  70.0   
6    14.0          4         454.0         220    4354           9.0  70.0   
..    ...        ...           ...         ...     ...           ...   ...   
387  27.0          1         140.0          86    2790          15.6  82.0   
388  44.0          1          97.0          52    2130          24.6  82.0   
389  32.0          1         135.0          84    2295          11.6  82.0   
390  28.0          1         120.0          79    2625          18.6  82.0   
391  31.0          1         119.0          82    2720          19.4  82.0   

    origin                       name  
0        1  chevrolet chevelle malibu  
1        1          buick skylark 320  
2        1         plymouth satellite  
3        1              amc rebel sst  
6        1           chevrolet impala  
..     ...                        ...  
387      1            ford mustang gl  
388      2                  vw pickup  
389      1              dodge rampage  
390      1                ford ranger  
391      1                 chevy s-10  

[389 rows x 9 columns]

Modify Columns


1.   Add mpg_high column with type categorical
2.   Delete mpg and name columns to prevent learning from mpg
3. Output Results





In [ ]:
avg_mpg = df['mpg'].mean()
print(avg_mpg)
df['mpg_high'] = (df['mpg'] > avg_mpg).astype('category').cat.codes

df = df.drop(columns=['mpg', 'name'])
df.head(5)

23.445918367346938


cylinders  displacement  horsepower  weight  acceleration  year origin  \
0          4         307.0         130    3504          12.0  70.0      1   
1          4         350.0         165    3693          11.5  70.0      1   
2          4         318.0         150    3436          11.0  70.0      1   
3          4         304.0         150    3433          12.0  70.0      1   
4          4         302.0         140    3449           NaN  70.0      1   

   mpg_high  
0         0  
1         0  
2         0  
3         0  
4         0

Data Expo w/ Graphs (seaborn)


*   catplot on mph_high column



In [ ]:
import seaborn as sns
import matplotlib.pyplot as plt


sns.catplot(x="mpg_high", y = "weight", hue = "cylinders", kind = "bar")

ValueError: ignored